.changeset/empty-zoos-beam.md

```diff
@@ -0,0 +1,7 @@
+---
+"wrangler": patch
+---
+
+fix: automatically drain incoming request bodies
+
+Previously, requests sent to `wrangler dev` with unconsumed bodies could result in `Network connection lost` errors. This change attempts to work around the issue by ensuring incoming request bodies are drained if they're not used. This is a temporary fix whilst we try to address the underlying issue. Whilst we don't think this change will introduce any other issues, it can be disabled by setting the `WRANGLER_DISABLE_REQUEST_BODY_DRAINING=true` environment variable. Note this fix is only applied if you've enabled Wrangler's bundling—`--no-bundle` mode continues to have the previous behaviour.
```

Comment:
I wonder if we should suggest that not draining bodies is bad practice generally? And that as a first call devs should attempt to do so rather than relying upon our workaround?

Reply:
It's not bad practice at all. If you have a POST endpoint which required auth, you will want to be able to return a 403 without reading the body at all

Reply:
Even in that case it would be better to cancel the body (if that worked, which it doesn't appear to)?
I think one reason why this is so tricky for the library/runtime author to resolve is that it is not always clear whether the user actually wants the body stream to be left open?

Reply:
This is a workaround for a workerd bug. The runtime team should fix the underlying bug. Do we even know what it is yet? Is it a memory leak of request body streams left open which eventually reach a limit?

If this is something that should be handled by users, the runtime team are gonna have to tell us that and really justify it.